Vize-Programming-Team/vize-meteor-app
# Feature - Docker Compose

I have written scripts that allow us to build and test the app with a single command. This works cross platform and the only programs you need to use this are Docker CE and Docker Compose. We will not need to install and set up any extra build tools to work with the new database system (technically you don't even need Meteor or NPM to make this work).

To test this in development mode:
1. Install Docker CE and Docker Compose.
1. Make sure the folders `meteor-app/node_modules` and `meteor-app/.meteor/local` exist. It does not matter what is in them.
1. In the root of the project run `sudo docker-compose -f docker-compose.dev.yml -f docker-compose.yml up`
1. Wait until you see a log message with "application running at...". This may take a few minutes the first time.
1. Go to http://localhost:3000/ just as we have in the past. You should see the site running.

WARNING: make sure you have a few spare GB of space on your hard drive. These tools use a little under 3GB of disk space.
